### DISCUSSION TITLE
Rpm compilation Cobbler3.3

### DISCUSSION THREAD

**Discussion by 1392273211** [cobbler/cobbler#3533] [Q&A]
How can Cobbler compile 3.3 into an rpm package and write a manual

**Comment by SchoolGuy** [comment]
We have this inside our installation guide: https://cobbler.readthedocs.io/en/latest/installation-guide.html

To summarize you probably want to use the multi-build since it only requires Docker on the host and builds the RPMs inside the container:

1. `git clone git@github.com:cobbler/cobbler.git`
2. `cd cobbler`
3. `docker-compose up -d`

A quote from our docs:

> After some time all containers expect one should be exited and you should see two new folders owned by root called rpm-build and deb-build.

**Reply by 1392273211** [reply]
Can the community provide documentation for directly pulling source code from virtualization or physical machines for compilation and installation, or compiling rpm

**Reply by SchoolGuy** [reply]
I don't understand what you mean with your sentence? What do you want to pull from where? Could you please try to rephrase that?

**Reply by 1392273211** [reply]
What I mean is that I tried to compile rpm but failed. Can the community directly provide compiled files?

**Reply by SchoolGuy** [reply]
Yes we actually do. You can take a look here: https://build.opensuse.org/package/show/systemsmanagement:cobbler:release33/cobbler

Choose your desired operating system and use your package manager to add the repository. After that, you can install the package.

**Reply by 1392273211** [reply]
Thanks, will it be compatible with other operating systems in the future? For example, Ali's Dragon Lizard, Tongxin Kirin v10 sp1 sp2 sp3 ctyunos

**Reply by SchoolGuy** [reply]
We do not build packages for those OSes and without community involvement I don't see that happening soon as I don't have access to those systems from Europe easily due to the language barrier that they impose for me.
